ext/storage_sources/azure_store/azure_connection.cpp

```diff
@@ -31,16 +31,38 @@
 #include <azure/core.hpp>
 #include <azure/storage/blobs.hpp>
 
+#include <stdlib.h>
+#include <iostream>
+#include <vector>
+#include <string>
+
 azure_connection::azure_connection(const std::string &bucket_name, const std::string &obj_prefix)
     : _azure_client(Azure::Storage::Blobs::BlobContainerClient::CreateFromConnectionString(
         std::getenv("AZURE_STORAGE_CONNECTION_STRING"), bucket_name)),
       _bucket_name(bucket_name), _object_prefix(obj_prefix)
 {
 }
 
+/*
+ * Lists all of the objects (blobs) on the bucket (container) utilising the listBlobs function form
+ * the Azure SDK
+ */
 int
-azure_connection::list_objects(std::vector<std::string> &objects) const
+azure_connection::list_objects(
+  const std::string &prefix, std::vector<std::string> &objects, bool list_single) const
 {
+    objects::clear();
+    
+    Azure::Storage::Blobs::ListBlobsOptions blob_parameters;
+    blob_parameters.Prefix = prefix;
```

Comment:
Have you tested this functionality?

Reply:
yes we've tested this functionality, ran them on Lachlan's machine with the put_objects function
![image](https://user-images.githubusercontent.com/98868895/212812499-60833d5f-8dcd-448a-b5ab-a34a3108ccaa.png)